app/(signed-in)/bookings/[bookingId]/page.tsx

```diff
@@ -1,22 +1,75 @@
 "use client";
 import { useParams, useSearchParams } from "next/navigation";
-import React, { useEffect } from "react";
+import React, { useEffect, useState } from "react";
+import {
+  PartialReservation,
+  retrieveSpecificReservation,
+} from "@/server-actions/reservation-actions";
+import { useSession } from "next-auth/react";
+import Link from "next/link";
 
 const Page = () => {
   const { bookingId: bookingIdSlug } = useParams();
   const searchParams = useSearchParams();
   const isBookingSuccessful = searchParams.get("success") === "true";
+  const { data: session, status } = useSession();
+  const [reservation, setReservation] = useState<PartialReservation | null>(
+    null
+  );
+  const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState(true);
 
   useEffect(() => {
-    // Retrieve booking information from DB.
-    // Call hoteldetails and hotelroom API for hotel information.
-  }, [bookingIdSlug, searchParams]);
+    const fetchReservation = async () => {
```

Comment:
These should be server actions. We want to hide the calls to our db and the functions to get to them. This introduces many risks.
Same with the one in `/bookings`